src/SushiBuildTaskProvider.ts

```diff
@@ -0,0 +1,36 @@
+import { Task, TaskGroup, TaskProvider, TaskScope, ShellExecution } from 'vscode';
+
+export class SushiBuildTaskProvider implements TaskProvider {
+  constructor() {}
+
+  provideTasks(): Task[] | undefined {
+    const sushiBuildTask = getSushiBuildTask();
+    const tasks = [sushiBuildTask];
+    return tasks;
+  }
+
+  resolveTask(_task: Task): Task | undefined {
+    const task = _task.definition.task;
+    if (task) {
+      const sushiBuildTask = getSushiBuildTask();
+      return sushiBuildTask;
+    }
+    return undefined;
+  }
+}
+
+export function getSushiBuildTask(): Task {
+  const sushiBuildTask = new Task(
+    { type: 'fsh', task: 'Build SUSHI' },
+    TaskScope.Workspace,
+    'sushi', // name
+    'fsh',
+    new ShellExecution('sushi ${workspaceFolder}'),
```

Comment:
I think that the input path needs to be quoted in order to work on windows filesystems.
```suggestion
    new ShellExecution('sushi "${workspaceFolder}"'),
```

Reply:
Great catch! Thank you!